Source/DafnyLanguageServer.Test/Util/ClientBasedLanguageServerTest.cs

```diff
@@ -212,14 +212,15 @@ protected ClientBasedLanguageServerTest(ITestOutputHelper output, LogLevel dafny
 
   public async Task<PublishDiagnosticsParams> GetLastDiagnosticsParams(TextDocumentItem documentItem, CancellationToken cancellationToken, bool allowStale = false) {
     var status = await WaitUntilAllStatusAreCompleted(documentItem, cancellationToken, allowStale);
-    var result = diagnosticsReceiver.History.LastOrDefault(d => d.Uri == documentItem.Uri);
+    var result = diagnosticsReceiver.GetLast(d => d.Uri == documentItem.Uri);
```

Comment:
There are two meanings of Last, it would be good to disambiguate them.
1. Last = the last available
1. Last = the one after which there will be none.

You could perhaps rename the first one as `LastAvailable`, and the second one as `Final`